modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2Tree.java

```diff
@@ -343,6 +362,16 @@ public H2Tree(
         created = initNew;
     }
 
+    private int maxAllowedInlineSize() {
+        // To avoid performance degradation, at least two items should fit into one page.
+        // So maximum payload size equals: P = (PS - H - 3L) / 2 - X , where P - Payload size, PS - page size, H - page
+        // header size, L - size of the child link, X - overhead per item.
+        int calculatedMaxItemSize = (pageMem.realPageSize(grpId) - BPlusIO.ITEMS_OFF - 3 * AbstractDataPageIO.LINK_SIZE)
+                / 2 - H2IOUtils.itemOverhead(mvccEnabled);
+
+        return Math.min(PageIO.MAX_PAYLOAD_SIZE, calculatedMaxItemSize);
```

Comment:
How is it possible that `calculatedMaxItemSize` will be greater than `MAX_PAYLOAD_SIZE`?